core/src/main/java/com/datastax/oss/driver/internal/core/channel/InFlightHandler.java

```diff
@@ -327,8 +328,13 @@ public void channelInactive(ChannelHandlerContext ctx) throws Exception {
 
   private ResponseCallback release(int streamId, ChannelHandlerContext ctx) {
     LOG.trace("[{}] Releasing stream id {}", logPrefix, streamId);
-    ResponseCallback responseCallback =
-        MoreObjects.firstNonNull(inFlight.remove(streamId), orphaned.remove(streamId));
+    ResponseCallback responseCallback;
+    if (!abortingAllRequests) {
+      responseCallback =
+          MoreObjects.firstNonNull(inFlight.remove(streamId), orphaned.remove(streamId));
+    } else {
+      responseCallback = orphaned.remove(streamId);
+    }
     orphanedSize = orphaned.size();
```

Comment:
This may not be the right way to handle this, but the problem I saw after the initial fix was that inflight.remove was being invoked while abort all inflight was being executed via the caughtException path.